scene/main/window.cpp

```diff
@@ -672,16 +672,21 @@ void Window::_event_callback(DisplayServer::WindowEvent p_event) {
 		case DisplayServer::WINDOW_EVENT_MOUSE_ENTER: {
 			_propagate_window_notification(this, NOTIFICATION_WM_MOUSE_ENTER);
 			Window *root = get_tree()->get_root();
-			DEV_ASSERT(!root->gui.windowmanager_window_over); // Entering a window while a window is hovered should never happen.
+			Window *previous_over = root->gui.windowmanager_window_over;
```

Comment:
If `mouse exit` was not received, `windowmanager_window_over` might be an invalid pointer to the deleted `Window` (the same might happen in `Viewport._update_mouse_over()`). Maybe `ObjectID` should be stored instead.

Reply:
I have tried multiple ways to cause a crash on X11 based on stale `gui.windowmanager_window_over`
- situations, that crashed without this fix
- explicitly freeing (queue_free and free)

But I wasn't able to cause a crash. However I can't rule out, that it is possible to cause a crash this way.